src/redis-benchmark.c

```diff
@@ -238,40 +240,56 @@ void _serverAssert(const char *estr, const char *file, int line) {
     *((char*)-1) = 'x';
 }
 
-static redisConfig *getRedisConfig(const char *ip, int port,
-                                   const char *hostsocket)
+static redisContext *getRedisContext(const char *ip, int port,
+                                     const char *hostsocket)
 {
-    redisConfig *cfg = zcalloc(sizeof(*cfg));
-    if (!cfg) return NULL;
-    redisContext *c = NULL;
-    redisReply *reply = NULL, *sub_reply = NULL;
+    redisContext *ctx = NULL;
+    redisReply *reply =  NULL;
     if (hostsocket == NULL)
-        c = redisConnect(ip, port);
+        ctx = redisConnect(ip, port);
     else
-        c = redisConnectUnix(hostsocket);
-    if (c == NULL || c->err) {
+        ctx = redisConnectUnix(hostsocket);
+    if (ctx == NULL || ctx->err) {
         fprintf(stderr,"Could not connect to Redis at ");
-        char *err = (c != NULL ? c->errstr : "");
+        char *err = (ctx != NULL ? ctx->errstr : "");
         if (hostsocket == NULL) fprintf(stderr,"%s:%d: %s\n",ip,port,err);
         else fprintf(stderr,"%s: %s\n",hostsocket,err);
-        goto fail;
+        return NULL;
     }
-
-    if(config.auth) {
-        void *authReply = NULL;
-        if (config.user == NULL)
-            redisAppendCommand(c, "AUTH %s", config.auth);
-        else
-            redisAppendCommand(c, "AUTH %s %s", config.user, config.auth);
-        if (REDIS_OK != redisGetReply(c, &authReply)) goto fail;
-        if (reply) freeReplyObject(reply);
-        reply = ((redisReply *) authReply);
-        if (reply->type == REDIS_REPLY_ERROR) {
-            fprintf(stderr, "ERROR: %s\n", reply->str);
-            goto fail;
+    if (config.auth == NULL) return ctx;
+    if (config.user == NULL)
+        reply = redisCommand(ctx,"AUTH %s", config.auth);
+    else
+        reply = redisCommand(ctx,"AUTH %s %s", config.user, config.auth);
+    if (reply != NULL) {
+        if (reply->type == REDIS_REPLY_ERROR){
+            if (hostsocket == NULL) fprintf(stderr, "Node %s:%d replied with error:\n%s\n",
+                                            ip, port, reply->str);
+            else fprintf(stderr, "Node %s replied with error:\n%s\n", hostsocket, reply->str);
+            return NULL;
```

Comment:
also here, you can use redisFree(ctx) and freeReplyObject(reply) before return to resolve this